File: src/main/java/cloud/common/RequestMessage.java
package cloud.common;

import instance.common.Request;
import se.sics.kompics.address.Address;
import se.sics.kompics.network.Message;

/**
 * 
 * @author Amir Moulavi
 * @date 2011-03-30
 *
 */

public class RequestMessage extends Message {

	private static final long serialVersionUID = 8682034389557178592L;
	private Request request;

	public RequestMessage(Address source, Address destination, Request request) {
		super(source, destination);
		this.request = request;
	}

	public Request getRequest() {
		return request;
	}

}

File: src/main/java/cloud/common/RequestMessage.scala
package cloud.common

import se.sics.kompics.network.Message
import se.sics.kompics.address.Address
import instance.common.Request
/**
 * @author Amir Moulavi
 * @date 2011-07-09
 *
 */
class RequestMessage(source:Address, destination:Address, val request:Request) extends Message(source, destination)